### DISCUSSION TITLE
Adapt custom llama2_chat tokenize to work with llama3

### DISCUSSION THREAD

**Discussion by LeDuySon** [axolotl-ai-cloud/axolotl#1723] [Q&A]
Hi guys, i have customized the https://github.com/OpenAccess-AI-Collective/axolotl/blob/main/src/axolotl/prompt_strategies/llama2_chat.py to work with my own chat template. Now i want to adapt it to llama3, what should i change to work with it? Right now, if i'm using the llama2_chat directly, it will have problem in the image below. Do you guys have any suggestions?
![image](https://github.com/OpenAccess-AI-Collective/axolotl/assets/33374938/8a47ec17-fe99-4048-afdf-9701302b1dfc)

**Comment by NanoCode012** [comment]
Hey, we just integrated better `chat_template` support that can allow you to simply swap the template when working with different models.

Looking at your issue however, did you perhaps reuse llama2 tokenizer for llama3?
